OrbitQt/ProcessItemModelTest.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <QAbstractItemModelTester>
+#include <QtGlobal>
+
+#include "ProcessItemModel.h"
+#include "gtest/gtest.h"
+#include "process.pb.h"
+
+namespace {
+
+class AssertNoQtLogWarnings {
+  static void MessageHandlerTest(QtMsgType type, const QMessageLogContext& context,
+                                 const QString& msg) {
+    static bool NO_WARNING_MSG = true;
+    QByteArray localMsg = msg.toLocal8Bit();
+    const char* file = context.file ? context.file : "";
+    const char* function = context.function ? context.function : "";
+    switch (type) {
+      case QtDebugMsg:
+        fprintf(stderr, "Debug: %s (%s:%u, %s)\n", localMsg.constData(), file, context.line,
+                function);
+        break;
+      case QtInfoMsg:
+        fprintf(stderr, "Info: %s (%s:%u, %s)\n", localMsg.constData(), file, context.line,
+                function);
+        break;
+      case QtWarningMsg:
+        EXPECT_EQ(false, NO_WARNING_MSG) << msg.toStdString();
+        break;
+      case QtCriticalMsg:
+        EXPECT_EQ(false, NO_WARNING_MSG) << msg.toStdString();
+        break;
+      case QtFatalMsg:
+        EXPECT_EQ(false, NO_WARNING_MSG) << msg.toStdString();
+        break;
+    }
+  }
+
+ public:
+  AssertNoQtLogWarnings() { qInstallMessageHandler(MessageHandlerTest); }
+
+  ~AssertNoQtLogWarnings() {
+    // Install default message handler
+    qInstallMessageHandler(nullptr);
+  }
+};
+
+}  // namespace
+
+namespace OrbitQt {
+
+TEST(ProcessItemModel, ProcessItemModel) {
+  AssertNoQtLogWarnings log_qt_test;
```

Comment:
This is very cool. Can you add a short comment why you're doing this? My guess is that the AbstractItemModelTester spits out warnings which have to be converted into a failed gtest asserts, right?

Reply:
I added a short comment, please let me know if insufficient, otherwise I will merge right away :)

Reply:
I'm sorry, I'm still missing the connection to QAbstractItemModelTester in here. Maybe instead of "produced by Qt"
you could write "produced by QAbstractItemModelTester". Technically that's not fully correct but I think it makes it easier
to understand why we want to fail on log messages.